### DISCUSSION TITLE
Value error

### DISCUSSION THREAD

**Discussion by allenmoench** [earthlab-education/Earth-Analytics-AY24#640] [Q&A]
Hello! Trying to figure out why the final model summary for the linear regression on my climate analysis won't print. I'm getting "ValueError: shapes (68,2) and (68,2) not aligned: 2 (dim 1) != 68 (dim 0)"

Anyone know what this is about? Here's the code: 

![image](https://github.com/user-attachments/assets/157c6ef0-8276-4603-b158-1ea682da0afc)

**Comment by eculler** [comment]
Hi @allenmoench, I will need some more information to help with this. I'm assuming you are using the `statmodels` library, and not `scikit-learn`? They behave a little differently, so make sure to include your imports in your post.

That said, I am guessing that the problem has to do with the shape of your data. Specifically, if you are using two predictor variables (as it looks like you are doing here), you may need to transpose them so that their dimensions are (2, 68) instead. The reason for this has to do with matrix multiplication, which is used in fitting statistical models, and which requires that the number of rows in the first matrix matches the number of columns in the second. These libraries usually handle 1D arrays pretty well, but need you to reshape 2D arrays. The easiest way to transpose is with `predictor.T`. Your observed values should be a 1D array no matter what.

Perhaps you are including extra columns in your observed and predicted variables? If that's the case, make sure that you get the individual column out of the DataFrame! It's ok to have an index, usually. Take a look at the values of `observed` and `predicted` to make sure that they look like what you expect.
